### ISSUE TITLE
Profile doesn't seem to save anything or appears to stays empty without error messages

### ISSUE THREAD

**Issue by vaab** [bluesky-social/pds#147]
My home PDS (here: https://bsky.0k.io ) was installed using the docker image and DNS setup is correct on a custom domain. It's a manual install : I had to read `install.sh` to reproduce the setup by hand on my home server. The docker image is behind an apache reverse proxy with no special settings.

My handle is @vaab.0k.io ([bsky profile page](https://bsky.app/profile/vaab.0k.io)), my DID is `did:plc:k5yxorcoacehrvfp7ams5fd2`, also:
- [bsky debug handle](https://bsky-debug.app/handle?handle=vaab.0k.io)
- [json output from my PDS about my profile](https://bsky.0k.io/xrpc/com.atproto.repo.describeRepo?repo=did:plc:k5yxorcoacehrvfp7ams5fd2)

I can create an invite code, use it to create an account, change handle, verify it, verify email.
Everything seems okay, sqlite files are being populated in the server side, I get lots of request logged on the server side, with all being 200 status.

I can consult posts, search for posts or people to follow, on the main page, and there are no obvious console errors.

**But nothing seems to be saved on my profile** (perhaps the writing side is okay, but the consultation is not), **my profile stays seemingly empty**:
- no follows despite me trying to follow accounts (no error is displayed when clicking on button following, even green checkmarks are displayed as if everything was successful),
- no profile photo seem to stick to my profile (no error is displayed, in console log I can have a `502 Upstream error` with a message: `pipethrough network error`... but that is not constant)

If I just reload, all these data are as if I have a pristine account with no data in it:

- people I followed will be still not displayed as being followed, I have no people in my list of account I follow
- my profile photo is still the default blue icon

Also, when clicking on my profile, I see InvalidHandle in place of my handle and I'm not sure this is related as this error is everywhere on the net even for people not having their own PDS.
In firefox dev console, I can see occasional (and probably unrelated ?):
```
Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://events.bsky.app/v2/initialize. (Reason: CORS request did not succeed). Status code: (null).
```
On the server side, this is what happens when I click on "Follow":
```
30 GET /xrpc/app.bsky.actor.getProfile?actor=did%3Aplc%3Acdzfvdl4pwp4pyw7atc52ewh > 200 request completed
30 GET /xrpc/app.bsky.actor.getPreferences > 304 request completed
30    > null [] loaded repo for
30    > null [] deleted indexed record
30    > null [] indexed record
30 POST /xrpc/com.atproto.repo.createRecord > 200 request completed
30 GET /xrpc/app.bsky.actor.getPreferences > 304 request completed
30 POST /xrpc/app.bsky.actor.putPreferences > 200 request completed
30 GET /xrpc/app.bsky.actor.getProfile?actor=did%3Aplc%3Ak5yxorcoacehrvfp7ams5fd2 > 200 request completed
30 GET /xrpc/app.bsky.actor.getPreferences > 200 request completed
30 GET /xrpc/app.bsky.graph.getSuggestedFollowsByActor?actor=did%3Aplc%3Acdzfvdl4pwp4pyw7atc52ewh > 200 request completed
```
Note: I'm using: `docker logs 0k_bluesky_1 --tail=10 -f | jq -r 'if .req then (.level | tostring) + " " + .req.method + " " + .req.url + " > " + (.res.statusCode | tostring) + " " +  .msg else (.level | tostring) + "    > " + (.status | tostring) + " [" + .message + "] " + .msg end'`

In one occasion, I got these logs on the server (like 20 times in a row), but this isn't what is typically happening after I click on the "Follow" button, but as the URL endpoint mentionned `subscribebRepos` and is a rare `401` error, it seemed possibly relevant to me):

```json
{
  "level": 50,
  "time": 1732177031971,
  "pid": 7,
  "hostname": "11b0e3bc6c66",
  "name": "AuthMissing",
  "status": 401,
  "message": "",
  "msg": "error in xrpc"
}
{
  "level": 30,
  "time": 1732177031972,
  "pid": 7,
  "hostname": "11b0e3bc6c66",
  "name": "pds",
  "req": {
    "id": 1196,
    "method": "GET",
    "url": "/xrpc/com.atproto.sync.subscribeRepos?cursor=0",
    "query": {
      "cursor": "0"
    },
    "params": {},
    "headers": {
      "host": "bsky.0k.io",
      "user-agent": "Go-http-client/1.1",
      "sec-websocket-key": "**REDACTED**",
      "sec-websocket-version": "13",
      "via": "1.1 bsky.0k.io",
      "x-forwarded-for": "209.249.133.122",
      "x-forwarded-host": "bsky.0k.io",
      "x-forwarded-server": "bsky.0k.io",
      "connection": "Keep-Alive"
    }
  },
  "res": {
    "statusCode": 401,
    "headers": {
      "x-powered-by": "Express",
      "access-control-allow-origin": "*",
      "cache-control": "private",
      "vary": "Authorization, Accept-Encoding",
      "content-type": "application/json; charset=utf-8",
      "content-length": "59",
      "etag": "W/\"3b-**REDACTED**\""
    }
  },
  "responseTime": 2,
  "msg": "request completed"
}
```
On the apache side this is the logs when this happens:
```
209.249.133.122 - - [21/Nov/2024:09:44:54 +0100] "GET /xrpc/com.atproto.server.describeServer HTTP/1.1" 200 3529 "-" "indigo/rev-d619a37"
67.213.161.34   - - [21/Nov/2024:09:44:55 +0100] "GET /xrpc/com.atproto.server.describeServer HTTP/1.1" 200 3529 "-" "indigo/rev-0e5ee80"
209.249.133.122 - - [21/Nov/2024:09:44:55 +0100] "GET /xrpc/com.atproto.sync.subscribeRepos?cursor=0 HTTP/1.1" 401 3518 "-" "Go-http-client/1.1"
67.213.161.34   - - [21/Nov/2024:09:44:55 +0100] "GET /xrpc/com.atproto.sync.subscribeRepos?cursor=0 HTTP/1.1" 401 3518 "-" "Go-http-client/1.1"
209.249.133.122 - - [21/Nov/2024:09:44:55 +0100] "GET /xrpc/com.atproto.server.describeServer HTTP/1.1" 200 3529 "-" "indigo/rev-de86564"
67.213.161.34   - - [21/Nov/2024:09:44:55 +0100] "GET /xrpc/com.atproto.sync.subscribeRepos?cursor=0 HTTP/1.1" 401 3518 "-" "Go-http-client/1.1"
209.249.133.122 - - [21/Nov/2024:09:44:56 +0100] "GET /xrpc/com.atproto.sync.subscribeRepos?cursor=0 HTTP/1.1" 401 3518 "-" "Go-http-client/1.1"
209.249.133.122 - - [21/Nov/2024:09:44:56 +0100] "GET /xrpc/com.atproto.sync.subscribeRepos?cursor=0 HTTP/1.1" 401 3518 "-" "Go-http-client/1.1"
67.213.161.34   - - [21/Nov/2024:09:44:56 +0100] "GET /xrpc/com.atproto.sync.subscribeRepos?cursor=0 HTTP/1.1" 401 3518 "-" "Go-http-client/1.1"
209.249.133.122 - - [21/Nov/2024:09:44:57 +0100] "GET /xrpc/com.atproto.sync.subscribeRepos?cursor=0 HTTP/1.1" 401 3518 "-" "Go-http-client/1.1"
67.213.161.34   - - [21/Nov/2024:09:44:57 +0100] "GET /xrpc/com.atproto.sync.subscribeRepos?cursor=0 HTTP/1.1" 401 3518 "-" "Go-http-client/1.1"
209.249.133.122 - - [21/Nov/2024:09:44:57 +0100] "GET /xrpc/com.atproto.sync.subscribeRepos?cursor=0 HTTP/1.1" 401 3518 "-" "Go-http-client/1.1"
209.249.133.122 - - [21/Nov/2024:09:44:57 +0100] "GET /xrpc/com.atproto.sync.subscribeRepos?cursor=0 HTTP/1.1" 401 3518 "-" "Go-http-client/1.1"
67.213.161.34   - - [21/Nov/2024:09:44:58 +0100] "GET /xrpc/com.atproto.sync.subscribeRepos?cursor=0 HTTP/1.1" 401 3518 "-" "Go-http-client/1.1"
```
We notice 2 IPs `209.249.133.122` and `67.213.161.34` (both originating from bluesky IPs)  trying to run the same `subscribeRepos` query and are not Authorized by my PDS.

I've tried docker image: `0.4.67` and `sha-b595125a28368fa52d12d3b6ca265c1bea06977f` which are the 2 most recent images at the time of this writing.

**Comment by pitterpatter22** [comment]
I am seeing the same error with nothing on profiles persisting (photo, posts, etc)

[Handle Debug](https://bsky-debug.app/handle?handle=pitter.bsky.pitterpatter.io)

### Version:
{"version":"0.4.67"}

### Json Description:
`{"handle":"pitter.bsky.pitterpatter.io","did":"did:plc:hxbti56u2k4vy3a3swxl6hv5","didDoc":{"@context":["https://www.w3.org/ns/did/v1","https://w3id.org/security/multikey/v1","https://w3id.org/security/suites/secp256k1-2019/v1"],"id":"did:plc:hxbti56u2k4vy3a3swxl6hv5","alsoKnownAs":["at://pitter.bsky.pitterpatter.io"],"verificationMethod":[{"id":"did:plc:hxbti56u2k4vy3a3swxl6hv5#atproto","type":"Multikey","controller":"did:plc:hxbti56u2k4vy3a3swxl6hv5","publicKeyMultibase":"zQ3shwWMx5pkaMNPC2VpuHqh7eZFF6sLjEGQEi3tPRiJVjHco"}],"service":[{"id":"#atproto_pds","type":"AtprotoPersonalDataServer","serviceEndpoint":"https://bsky.pitterpatter.io"}]},"collections":["app.bsky.actor.profile","app.bsky.feed.post","app.bsky.graph.follow"],"handleIsCorrect":true}`


### Error Messages in Javascript Console:
```
[Info] Successfully preconnected to https://bsky.social/
[Error] Origin https://bsky.app is not allowed by Access-Control-Allow-Origin. Status code: 200
[Error] Fetch API cannot load https://statsigapi.net/v1/sdk_exception due to access control checks.
[Error] Failed to load resource: Origin https://bsky.app is not allowed by Access-Control-Allow-Origin. Status code: 200 (sdk_exception, line 0)
[Error] The network connection was lost.
[Error] Fetch API cannot load https://events.bsky.app/v2/initialize due to access control checks.
[Error] Failed to load resource: The network connection was lost. (initialize, line 0)
[Info] Successfully preconnected to https://bsky.network/
[Error] The network connection was lost.
[Error] Fetch API cannot load https://events.bsky.app/v2/rgstr due to access control checks.
[Error] Failed to load resource: The network connection was lost. (rgstr, line 0)
[Error] The network connection was lost.
[Error] Fetch API cannot load https://events.bsky.app/v2/initialize due to access control checks.
[Error] Failed to load resource: The network connection was lost. (initialize, line 0)
[Error] The network connection was lost.
[Error] Fetch API cannot load https://events.bsky.app/v2/rgstr due to access control checks.
[Error] Failed to load resource: The network connection was lost. (rgstr, line 0)
[Error] The network connection was lost.
[Error] Fetch API cannot load https://events.bsky.app/v2/initialize due to access control checks.
[Error] Failed to load resource: The network connection was lost. (initialize, line 0)
[Error] The network connection was lost.
[Error] Fetch API cannot load https://events.bsky.app/v2/rgstr due to access control checks.
[Error] Failed to load resource: The network connection was lost. (rgstr, line 0)
[Error] Origin https://bsky.app is not allowed by Access-Control-Allow-Origin. Status code: 200
[Error] Fetch API cannot load https://statsigapi.net/v1/sdk_exception due to access control checks.
[Error] Failed to load resource: Origin https://bsky.app is not allowed by Access-Control-Allow-Origin. Status code: 200 (sdk_exception, line 0)
[Error] The network connection was lost.
[Error] Fetch API cannot load https://events.bsky.app/v2/rgstr due to access control checks.
[Error] Failed to load resource: The network connection was lost. (rgstr, line 0)
[Error] The network connection was lost.
[Error] Fetch API cannot load https://events.bsky.app/v2/rgstr due to access control checks.
[Error] Failed to load resource: The network connection was lost. (rgstr, line 0)
[Error] The network connection was lost.
```

**Comment by colinrgodsey** [comment]
Also having the same issue with my PDS. Oddly enough, my test account on my pds works fine, but my main account doesn't. This also happened spontaneously in the middle of the day yesterday. The pds was not altered or touched at that time.

**Comment by colinrgodsey** [comment]
alright.... I had to rebuild my repo using this shady looking script: https://github.com/bluesky-social/atproto/pull/2528/files

but it worked

**Comment by kevindqc** [comment]
How do you invoke the rebuild-repo script?

**Comment by colinrgodsey** [comment]
@kevindqc was a bit weird to do with my setup, but basically:

* Copy `run-script.js` ([here](https://github.com/bluesky-social/atproto/pull/2528/files#diff-b4239e7657087f4db4a9b4f970e3ec2c454eb8d1b0a8e4e0e6ee52d89a3ff21a)) to the root of my PDS volume for the container
Then in the container:
* `cd /pds` (or whatever your mount point is in the container for the volume)
* `ln -s /app/node_modules` (just symlinks the install node environment to `/pds`)
* `node run-script.js rebuild-repo did:plc:gcienmzzp3sbfurcnkqtcx4m` (using your DID of course)

You could also run it locally, but you need to clone https://github.com/bluesky-social/atproto and set up your env vars. I ran it in the container just so I could skip that.

**Comment by kevindqc** [comment]
Thanks, this worked great! I also ran it from the container.
```bash
root@ubuntu-s-1vcpu-1gb-tor1-01:~# docker ps
CONTAINER ID   IMAGE                            COMMAND                  CREATED       STATUS                 PORTS     NAMES
2e8a3ae8d760   caddy:2                          "caddy run --config …"   3 hours ago   Up 3 hours                       caddy
103636564c7e   containrrr/watchtower:latest     "/watchtower"            3 hours ago   Up 3 hours (healthy)             watchtower
96921727fa9c   ghcr.io/bluesky-social/pds:0.4   "dumb-init -- node -…"   3 hours ago   Up 3 hours                       pds

root@ubuntu-s-1vcpu-1gb-tor1-01:~# docker exec -it 96921727fa9c sh
/app # cd /pds
/pds # ln -s /app/node_modules
/pds # vi run-script.js
/pds # chmod +x run-script.js 
/pds # node run-script.js rebuild-repo did:plc:hkqwq2vufoj3ljrow2rewufy
Record count:  3
Existing blocks:  7
Deleting blocks: 1
Adding blocks:  1
Continue? y/n 
DONE

```

**Comment by pitterpatter22** [comment]
That worked for me too, everything instantly showed up. Thanks for saving me hours of troubleshooting!

**Comment by colinrgodsey** [comment]
ya'll are welcome to follow me on my further adventures in breaking the bluesky PDS: @colingodsey

(I will eventually have a good set of reports to add here)

**Comment by vl-odoo** [comment]
I'm still stuck despite having run the promising fix of @colinrgodsey ! 

But, to be noted, it seems the script will trigger the salve of `/xrpc/com.atproto.sync.subscribeRepos?cursor=0` requests all being replied a `401` AuthMissing . They are refused because they don't have "authorization" key/value in their headers here: https://github.com/bluesky-social/atproto/blob/main/packages/pds/src/auth-verifier.ts#L417 . I don't get that because, according to [auth doc](https://docs.bsky.app/docs/advanced-guides/api-directory), I quote:
 
> data updates from the entire network can be streamed over a WebSocket using the `com.atproto.sync.subscribeRepos` Lexicon endpoint. This endpoint does not require auth, and can be made to individual PDS instances (for data just from that PDS), or to a Relay to receive updates from the entire network.

I feel I'll need to have a look about the websocket configuration of my proxy... this will be for another day.

**Comment by vaab** [comment]
@femxiii thanks, I didn't see your answer until now. I have done the same on my apache proxy:

```
        RewriteEngine On
        RewriteCond %{HTTP:Upgrade} ^websocket$ [NC]
        RewriteCond %{HTTP:Connection} Upgrade [NC]
        RewriteRule /(.*)$ ws://bluesky:3000/\$1 [P,L]
```
Where `bluesky:3000` is the host and port where the docker container resides.

This solves the `401 AuthMissing` errors on my side on the requests that comes from bluesky company's IP towards my PDS, but it doesn't solve my main problem : my profile is still stuck and stay's empty unfortunately.

**Comment by vaab** [comment]
I managed to solve the "Invalid Handle" issue, after some trial and error, it seems by preventing firefox to enforce strict CORS (go to `about:config` , search for `security.fileuri.strict_origin_policy` and switch to `false`). 

Alas, this did not improve my issue about the profile being stuck and showing me an empty profile.

The sqlite database `record` table is updated in ~actors/4c/did:plc:k5yxorcoacehrvfp7ams5fd2/store.sqlite~ each time.

Got it ! I overlooked (my bad!) the `wsdump` check documented on the front README of this project, and assumed that if requests where now getting 200 ok status, that meant my websocket configuration was not the culprit. I was wrong, using:

```
go install github.com/nrxr/wsdump@latest
wsdump "wss://bsky.0k.io/xrpc/com.atproto.sync.subscribeRepos?cursor=0"
```

Helped me show a problem: I got `panic: malformed HTTP status code "HTML"`. Using `tcpdump` on the docker container side, I saw a mistake in my apache configuration of the service. I corrected it in the previous post in this thread to avoid spreading a bad configuration (it was a spurious `\` in front of the final `$`).

I then applied the `run-script` as described in [this previous message](https://github.com/bluesky-social/pds/issues/147#issuecomment-2491927352) from @colinrgodsey ... and everything appeared.

So if anyone is proxying through apache, here's the working conf:

```
        RewriteEngine On
        RewriteCond %{HTTP:Upgrade} ^websocket$ [NC]
        RewriteCond %{HTTP:Connection} Upgrade [NC]
        RewriteRule /(.*)$ ws://bluesky:3000/$1 [P,L]
```

Of course, replace `bluesky` by the IP or name that can be resolved to your bluesky's PDS docker container.
